### DISCUSSION TITLE
How to know if isValid has been validated/updated?

### DISCUSSION THREAD

**Discussion by jpsantos7** [react-hook-form/react-hook-form#10319] [Q&A] [answered]
How can I be certain that `isValid` has already been updated?

I need to run a function only if the form is actually valid but because the useForm is async if I check for `isValid` it returns the previous value because it hasn't been updated yet.

Any help would be appreciated 🙏

**Comment by leapful** [comment, marked as answer]
@jpsantos7 

You can use `useEffect` with `formState.isValid`.

For example:

```
const { formState: { isValid } } = useForm();

useEffect(() => {
   if (isValid) {
      // Execute any function
   }
}, [isValid])

```

**Reply by jpsantos7** [reply]
Thanks @leapful
